### ISSUE TITLE
Show unmapped percussion notes in red

### ISSUE THREAD

**Issue by zacjansheski** [musescore/MuseScore#25712]
### Your idea

In the same manner that notes that are out of the range of a tonal instrument are shown in red, a note that is not mapped on a percussion instrument should also be displayed in red to signal to the user that it is not mapped

### Problem to be solved

This will signal to the user that there is an issue in the instance escribed here: https://github.com/musescore/MuseScore/issues/14016#issuecomment-2514835183

### Checklist

- [X] This request follows the [guidelines](https://github.com/musescore/MuseScore/wiki/Reporting-bugs-and-issues) for reporting issues
- [X] I have verified that this feature request has not been logged before, by searching the [issue tracker](https://github.com/musescore/MuseScore/issues) for similar requests

**Comment by avvvvve** [comment]
@bkunda Just to make sure I understand correctly: When we fix #14016 for 4.5, when changing a sound either from Muse Sounds Drum Kit to an MS Basic drumset or vice versa, we want the notation to automatically change to _correct_ new notes (and not wait for the user to input a note to do so), right? The idea being that we would never show an unmapped note in the score, so there's no need for color-coding.

I don't know that we should be more specific than saying something like 'percussion notation and playback may be affected' since it seems pretty unpredictable what happens now. Here's how I'd propose we rework the existing 'Are you sure you want to change sounds?' dialog:

![Image](https://github.com/user-attachments/assets/29b9bcc3-1c9e-4abb-9193-5b1b9b9731ae)

Changes:

- Show sound flag and/or percussion messaging dependent on what sound(s) are being changed
     - Question: In-app, are we only putting a period at the end of the last bullet point in a bulleted list?
- Add a tip to duplicate instruments first to avoid losing sound flags/faulty percussion notation
- Change the 'Change sound' CTA to 'Confirm change' so we don't have to have separate buttons for singular/plural 'Change sound'/'Change sounds' (that's how it's currently built, so I'm trying to cut back on the number of permutations needed)

For context, here are the existing dialogs for sound flags:

<img width="1269" alt="Image" src="https://github.com/user-attachments/assets/4a474698-0ce8-41b8-95b0-ee4967dd12da" />

**Comment by zacjansheski** [comment]
We should have some messaging whenever a new drm is loaded at the very least.

I'm still in favor of showing unmapped notes in red.  All notes are technically "in the playable range" because unpitched percussion instruments have no range.  I think showing the unmapped notes in a different color would make it easier for the user to make adjustments when changing from one sound library instrument/mapping to another.

**Comment by erinic04** [comment]
Yes, some sort of visual difference for unmapped notes would be so helpful for what I do.
For instance, I import a midi file with a drumset, I paste the part onto the staff, and then I make sure all the pitches are mapped in the right place.
However, when there is a note mapped to the top line of the staff, it shares a space with all the unmapped notes, so I have to temporarily move that mapped note out of the way to be able to tell which notes are unmapped.

**Comment by marnen** [comment]
We show harp notes that aren’t playable with the current pedal settings in red too. This seems consistent with that.
